lib/msf/core/data_store.rb

```diff
@@ -169,29 +169,29 @@ def to_s(delim = ' ')
   def to_h
     datastore_hash = {}
     self.keys.each do |k|
-      datastore_hash[k.to_s] = self[k].to_s
+      datastore_hash[k.to_s.dup.force_encoding('UTF-8')] = self[k].to_s.dup.force_encoding('UTF-8')
     end
     datastore_hash
   end
 
   # Hack on a hack for the external modules
-  def to_nested_values
```

Comment:
We traced down the original use of this method to https://github.com/rapid7/metasploit-framework/pull/10002 which allows datastore values to be a nested array, which I'm not sure is something we'd want